tracex_project/extraction/tests/test_views.py

```diff
@@ -0,0 +1,390 @@
+        """Test that requesting a patient journey that does not exist returns a 404 error."""
+        response = self.client.get(reverse('journey_details', kwargs={'pk': 999}))
+
+        self.assertEqual(response.status_code, 404)
+
+    def test_post_method_redirect(self):
+        """Test that a POST request to the view redirects to the same page."""
+        # Perform a GET request to set up session data
+        response = self.client.get(self.url)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(self.client.session['patient_journey_id'], self.mock_journey.id)
+
+        # Perform a POST request to the same view
+        response = self.client.post(self.url)
+
+        self.assertEqual(response.status_code, 302)
+        self.assertRedirects(response, reverse('journey_filter'))
+
+
+class JourneyFilterViewTests(TestCase):
+    """Test cases for the JourneyFilterView."""
+
+    fixtures = ["tracex_project/extraction/fixtures/prompts_fixture.json"]
+
+    @staticmethod
+    def dummy_get_response():
+        """This is a dummy function to satisfy the get_response parameter of the SessionMiddleware."""
+        return None
+
+    def setUp(self):  # pylint: disable=invalid-name
+        """Set up method that gets called everytime before tests are executed."""
+        self.client = Client()
+        self.mock_journey = PatientJourney.manager.create(name='Test Journey',
+                                                          patient_journey='This is a test patient journey.')
+        self.url = reverse('journey_filter')
+        self.factory = RequestFactory()
+
+    def test_view_get_request(self):
+        """Tests that the view URL exists and is accessible by passing a GET request."""
+        response = self.client.get(self.url)
+        resolver = resolve(self.url)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(resolver.func.view_class, JourneyFilterView)
+
+    def test_uses_correct_template(self):
+        """Tests that the view uses the correct template."""
+        response = self.client.get(self.url)
+
+        self.assertTemplateUsed(response, 'filter_journey.html')
+
+    def test_view_uses_correct_form(self):
+        """Tests that the view uses the correct form."""
+        response = self.client.get(self.url)
+
+        self.assertIsInstance(response.context['form'], FilterForm)
+
+    def test_get_context_data(self):
+        """Test that the `is_comparing` context variable is added correctly in the `get_context_data` method."""
+        request = self.factory.get(self.url)
+        request.session = {}
+        view = JourneyFilterView()
+        view.request = request
+        context = view.get_context_data()
+
+        self.assertIn('is_comparing', context)
+
+    # Non-deterministic test since orchestrator is executed
+    def test_form_valid(self):
+        """Test that a valid form submission redirects to the correct URL."""
```

Comment:
"""Test that a valid form submission redirects to the correct URL."""
```suggestion
        """Tests that a valid form submission redirects to the correct URL."""
```